libavcodec/vvc/inter_template.c

```diff
@@ -455,25 +464,35 @@ static void FUNC(apply_bdof_min_block)(pixel* dst, const ptrdiff_t dst_stride, c
 static void FUNC(apply_bdof)(uint8_t *_dst, const ptrdiff_t _dst_stride, const int16_t *_src0, const int16_t *_src1,
     const int block_w, const int block_h)
 {
-    int16_t gradient_h[2][BDOF_BLOCK_SIZE * BDOF_BLOCK_SIZE];
-    int16_t gradient_v[2][BDOF_BLOCK_SIZE * BDOF_BLOCK_SIZE];
+    tpixel gradient_h[2][BDOF_BLOCK_SIZE * BDOF_BLOCK_SIZE];
+    tpixel gradient_v[2][BDOF_BLOCK_SIZE * BDOF_BLOCK_SIZE];
     int vx, vy;
     const ptrdiff_t dst_stride  = _dst_stride / sizeof(pixel);
     pixel* dst                  = (pixel*)_dst;
+    const int ps                = BIT_DEPTH >> 3;
+
+    int16_t *gh0 = (int16_t *)gradient_h[0];
```

Comment:
prof_grad_filter is a local function, so we do not need to covert it to int16 _t, just use tpixel.